src/test/util/mining.cpp

```diff
@@ -74,10 +74,12 @@ CTxIn MineBlock(const NodeContext& node, const CScript& coinbase_scriptPubKey)
     return CTxIn{block->vtx[0]->GetHash(), 0};
 }
 
-std::shared_ptr<CBlock> PrepareBlock(const NodeContext& node, const CScript& coinbase_scriptPubKey)
+std::shared_ptr<CBlock> PrepareBlock(const NodeContext& node, const CScript& coinbase_scriptPubKey, bool check)
```

Comment:
nit: Could pass in the options struct to avoid having to change this again if another option is added?
Could also add a redirect function without the argument, similar to BlockAssembler, to avoid having to change existing call sites?

Feel free to ignore, but then you could use clang-tidy named-args to at least document the bool literals.

Reply:
Good idea, did this instead of the bool